From a26afd10bc72fcfb0862272aa0d16f25d7a0d640 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Tue, 10 Oct 2017 01:38:11 +0000
Subject: [PATCH] MORAY-438 Documentation lists NoDatabaseError instead of
 NoDatabasePeersError MORAY-304 Remove numWorkers from all documentation and
 configs/templates Reviewed by: Jared Morrow <jm@joyent.com> Approved by:
 Jared Morrow <jm@joyent.com>

---
 docs/index.md                     | 22 +++++++++++-----------
 etc/config.coal.manta.json        |  1 -
 etc/config.lab.json               |  1 -
 lib/errors.js                     |  9 ---------
 sapi_manifests/moray/template     |  1 -
 sdc/sapi_manifests/moray/template |  1 -
 6 files changed, 11 insertions(+), 24 deletions(-)

diff --git a/docs/index.md b/docs/index.md
index 91e1e06..59c9f7a 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -149,7 +149,7 @@ A "fully loaded" config (without post triggers) would look like this:
 * `InvalidBucketConfigError`
 * `InvalidBucketNameError`
 * `InvalidIndexDefinitionError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 * `NotFunctionError`
 
 Plus any currently unhandled Postgres Errors (such as relation already exists).
@@ -190,7 +190,7 @@ besides the name of the bucket you want to fetch (and of course request\_id).
 ### Errors
 
 * `BucketNotFoundError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 
 ### CLI
 
@@ -250,7 +250,7 @@ Returns the configuration for all buckets.
 
 ### Errors
 
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 
 ### CLI
 
@@ -351,7 +351,7 @@ Or alternatively:
 * `BucketNotFoundError`
 * `InvalidBucketConfigError`
 * `InvalidIndexDefinitionError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 * `NotFunctionError`
 
 ### CLI
@@ -380,7 +380,7 @@ Deletes a bucket, *and all data in that bucket!*
 ### Errors
 
 * `BucketNotFoundError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 
 ### CLI
 
@@ -446,7 +446,7 @@ looks like:
 
 * `BucketNotFoundError`
 * `EtagConflictError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 * `UniqueAttributeError`
 * `InvalidIndexTypeError`
 
@@ -496,7 +496,7 @@ for a full example.
 
 * `BucketNotFoundError`
 * `ObjectNotFoundError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 
 Plus any currently unhandled Postgres Errors (such as relation already exists).
 
@@ -666,7 +666,7 @@ PostgreSQL resources remain in use until the query ultimately does complete.
 
 * `BucketNotFoundError`
 * `InvalidQueryError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 * `NotIndexedError`
 * `InvalidIndexTypeError`
 
@@ -740,7 +740,7 @@ options to get `test/set` semantics.
 * `BucketNotFoundError`
 * `EtagConflictError`
 * `ObjectNotFoundError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 
 Plus any currently unhandled Postgres Errors (such as relation already exists).
 
@@ -835,7 +835,7 @@ the same bucket but it's likely to race on rows and incur rollbacks/slowdowns.
 ### Errors
 
 * `BucketNotFoundError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 
 Plus any currently unhandled Postgres Errors (such as relation already exists).
 
@@ -920,7 +920,7 @@ The default operation is `put`.
 
 * `BucketNotFoundError`
 * `EtagConflictError`
-* `NoDatabaseError`
+* `NoDatabasePeersError`
 * `UniqueAttributeError`
 
 Plus any currently unhandled Postgres Errors (such as relation already exists).
diff --git a/etc/config.coal.manta.json b/etc/config.coal.manta.json
index 5320d54..5790d98 100644
--- a/etc/config.coal.manta.json
+++ b/etc/config.coal.manta.json
@@ -6,7 +6,6 @@
       "type": "udp"
     }
   },
-  "numWorkers": 0,
   "manatee": {
     "manatee": {
       "path": "/manatee/1.moray.coal.joyent.us/election",
diff --git a/etc/config.lab.json b/etc/config.lab.json
index 3bf9fcc..7e2986e 100644
--- a/etc/config.lab.json
+++ b/etc/config.lab.json
@@ -7,7 +7,6 @@
                         "type": "udp"
                 }
         },
-        "numWorkers": 0,
         "manatee": {
                 "path": "/manatee/sdc/election",
                 "pg": {
diff --git a/lib/errors.js b/lib/errors.js
index d1bb039..93be507 100644
--- a/lib/errors.js
+++ b/lib/errors.js
@@ -214,14 +214,6 @@ function InvocationError() {
 util.inherits(InvocationError, VError);
 
 
-function NoDatabaseError(cause) {
-    WError.call(this, (cause || {}),
-                'Not connected to manatee and/or postgres');
-    this.name = this.constructor.name;
-}
-util.inherits(NoDatabaseError, WError);
-
-
 function NoDatabasePeersError(msg) {
     assert.string(msg, 'msg');
     WError.call(this, msg);
@@ -364,7 +356,6 @@ module.exports = {
     InvalidRequireError: InvalidRequireError,
     InvalidQueryError: InvalidQueryError,
     InvocationError: InvocationError,
-    NoDatabaseError: NoDatabaseError,
     NoDatabasePeersError: NoDatabasePeersError,
     NotFunctionError: NotFunctionError,
     NotIndexedError: NotIndexedError,
diff --git a/sapi_manifests/moray/template b/sapi_manifests/moray/template
index 97ce216..00b9609 100644
--- a/sapi_manifests/moray/template
+++ b/sapi_manifests/moray/template
@@ -6,7 +6,6 @@
       "type": "udp"
     }
   },
-  "numWorkers": 0,
   "manatee": {
     "manatee": {
       "path": "/manatee/{{SERVICE_NAME}}",
diff --git a/sdc/sapi_manifests/moray/template b/sdc/sapi_manifests/moray/template
index 991ccbe..28349ce 100644
--- a/sdc/sapi_manifests/moray/template
+++ b/sdc/sapi_manifests/moray/template
@@ -6,7 +6,6 @@
         "type": "udp"
       }
   },
-  "numWorkers": 0,
   "manatee": {
     "manatee":{
       "path": "/manatee/{{manatee_shard}}",
-- 
2.21.0

